app/utils/types.ts

```diff
@@ -187,10 +189,15 @@ export enum TransactionKindId {
     Transfer_with_schedule = 19,
     Update_credentials = 20,
     Register_data = 21,
+    Simple_transfer_with_memo = 22,
+    Encrypted_transfer_with_memo = 23,
+    Transfer_with_schedule_and_memo = 24,
 }
+
 export interface SimpleTransferPayload {
     amount: string;
     toAddress: string;
+    memo?: string;
```

Comment:
The typing is better if you make a new interface:
```
export interface SimpleTransferWithMemoPayload extends SimpleTransferPayload {
    memo: string;
}
```
Leaving it as optional on the regular simple transfer is not correct as it cannot have a memo.

Reply:
And after writing this I saw `SimpleTransferWithMemo`, but then I don't understand why you need an optional `memo` here.

Reply:
I have refactored it to make the transaction more separate.